File: app/src/main/java/com/panda/ap_anime/anmi_details/details_repo/StreamObject.java
package com.panda.ap_anime.anmi_details.details_repo;

public class StreamObject {

    public String getDomainBame() {
        return domainBame;
    }

    public void setDomainBame(String domainBame) {
        this.domainBame = domainBame;
    }

    public String getQuality() {
        return quality;
    }

    public void setQuality(String quality) {
        this.quality = quality;
    }

    public String getStreamname() {
        return streamname;
    }

    public void setStreamname(String streamname) {
        this.streamname = streamname;
    }


    public StreamObject() {

    }

    public StreamObject(String dname, String quality, String streamname) {
        this.streamname = streamname;
        this.quality = quality;
        this.domainBame = dname;
    }

    private String domainBame;
    private String quality;
    private String streamname;


}

File: app/src/main/java/com/panda/ap_anime/anmi_details/details_repo/StreamObject.kt
package com.panda.ap_anime.anmi_details.details_repo

import android.support.v7.util.DiffUtil

class StreamObject(dname: String, quality: String, streamname: String) {

    var domainBame: String? = dname
    var quality: String? = quality
    var streamname: String? = streamname


    override fun equals(other: Any?): Boolean {
        if (other is StreamObject && other === this)
            return true

        val user = other as StreamObject?

        return user!!.streamname == this.streamname && user.domainBame == this.domainBame
    }

    override fun hashCode(): Int {
        var result = domainBame?.hashCode() ?: 0
        result = 31 * result + (quality?.hashCode() ?: 0)
        result = 31 * result + (streamname?.hashCode() ?: 0)
        return result
    }


    companion object {

        var DIFF_CALLBACK: DiffUtil.ItemCallback<StreamObject> = object : DiffUtil.ItemCallback<StreamObject>() {
            override fun areItemsTheSame(oldItem: StreamObject, newItem: StreamObject): Boolean {
                return oldItem.streamname == newItem.streamname
            }

            override fun areContentsTheSame(oldItem: StreamObject, newItem: StreamObject): Boolean {
                return oldItem == newItem
            }
        }
    }
}
